website/js/index.js

```diff
@@ -559,15 +577,9 @@ var VariantDetail = React.createClass({
         });
         localStorage.setItem("collapse-group_" + groupTitle, willBeCollapsed);
 
-        // defer re-layout until the state change has completed
-        const me = this;
-
-        setTimeout(() => {
-            // this forces a re-render after a group has expanded/collapsed, fixing the layout
-            // note that 300ms just happens to be the duration of the expand/collapse animation
-            // it'd be better to run the re-layout whenever the animation ends
-            me.forceUpdate();
-        }, 300);
+        // find the actual collapsing DOM element so we can attach a handler to its transition-end event
+        const collapser = collapsingElemParent.parentElement.parentElement.getElementsByClassName("panel-collapse")[0];
```

Comment:
is this our best approach to finding the element? not a huge deal but looks fragile at a glance

Reply:
Agreed, it's definitely flimsy. What this is effectively doing is getting refs for all the <Panels> that I use without actually having to add that into each one of them (which I was concerned would be a lot of work to pass back from the new tile types), but I ended up writing the code to pass back refs for those tile types anyway later on...

So, good catch there. 👍I'll rewrite it to use proper refs to the Panels as it should. It's late here at the moment, but expect at least one more commit tomorrow that implements that.

Reply:
Sounds good! No hurry, whenever you're ready :)

Reply:
FYI, I opted to have each panel that calls onChangeGroupVisibility pass a ref to the header, which is way cleaner than attempting to find it from the event target. It's included in the latest commit.